src/main/osd/osd.c

```diff
@@ -687,17 +688,21 @@ static bool osdDisplayStat(int statistic, uint8_t displayRow)
 
     case OSD_STAT_MIN_BATTERY:
         tfp_sprintf(buff, "%d.%02d%c", stats.min_voltage / 100, stats.min_voltage % 100, SYM_VOLT);
-        osdDisplayStatisticLabel(displayRow, "MIN BATTERY", buff);
+        osdDisplayStatisticLabel(displayRow, osdConfig()->stat_show_cell_value? "MIN AVG CELL" : "MIN BATTERY", buff);
         return true;
 
     case OSD_STAT_END_BATTERY:
         tfp_sprintf(buff, "%d.%02d%c", stats.end_voltage / 100, stats.end_voltage % 100, SYM_VOLT);
-        osdDisplayStatisticLabel(displayRow, "END BATTERY", buff);
+        osdDisplayStatisticLabel(displayRow, osdConfig()->stat_show_cell_value ? "END AVG CELL" : "END BATTERY", buff);
         return true;
 
     case OSD_STAT_BATTERY:
-        tfp_sprintf(buff, "%d.%02d%c", getBatteryVoltage() / 100, getBatteryVoltage() % 100, SYM_VOLT);
-        osdDisplayStatisticLabel(displayRow, "BATTERY", buff);
+        if (osdConfig()->stat_show_cell_value) {
+            tfp_sprintf(buff, "%d.%02d%c", getBatteryAverageCellVoltage() / 100, getBatteryAverageCellVoltage() % 100, SYM_VOLT);
+        } else {
+            tfp_sprintf(buff, "%d.%02d%c", getBatteryVoltage() / 100, getBatteryVoltage() % 100, SYM_VOLT);
+        }
```

Comment:
This way would save a bit of flash:
```
    const uint16_t statsVoltage = osdConfig()->stat_show_cell_value ? getBatteryAverageCellVoltage() : getBatteryVoltage();
    tfp_sprintf(buff, "%d.%02d%c", statsVoltage / 100, statsVoltage % 100, SYM_VOLT);
```

Reply:
Fully agreed on the recommendation, as it will make the code easier to maintain as well.
I suspect that gcc might pick up on the two invocations of the same function with only one parameter expression changed, and optimise if left to its own devices. :wink:

Reply:
Agreed. Working on it.

Reply:
Use `git commit --amend` for easy addition